contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AggregatorValidatorInterface} from "../../../shared/interfaces/AggregatorValidatorInterface.sol";
+import {TypeAndVersionInterface} from "../../../interfaces/TypeAndVersionInterface.sol";
+import {ZKSyncSequencerUptimeFeedInterface} from "./../interfaces/ZKSyncSequencerUptimeFeedInterface.sol";
+
+import {SimpleWriteAccessController} from "../../../shared/access/SimpleWriteAccessController.sol";
+
+import {IBridgehub, L2TransactionRequestDirect} from "@zksync/contracts/l1-contracts/contracts/bridgehub/IBridgehub.sol";
+
+///
+/// @title ZKSyncValidator - makes cross chain call to update the Sequencer Uptime Feed on L2
+///
+contract ZKSyncValidator is TypeAndVersionInterface, AggregatorValidatorInterface, SimpleWriteAccessController {
+  int256 private constant ANSWER_SEQ_OFFLINE = 1;
+  uint32 private s_gasLimit;
+  uint32 private s_l2GasPerPubdataByteLimit;
+
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  address public immutable L1_CROSS_DOMAIN_MESSENGER_ADDRESS;
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  address public immutable L2_UPTIME_FEED_ADDR;
+
+  uint32 private immutable i_chainId;
+  uint32 private constant TEST_NET_CHAIN_ID = 300;
+  uint32 private constant MAIN_NET_CHAIN_ID = 324;
+
+  ///
+  /// @notice emitted when gas cost to spend on L2 is updated
+  /// @param gasLimit updated gas cost
+  ///
+  event GasLimitUpdated(uint32 gasLimit);
+
+  ///
+  /// @param l1CrossDomainMessengerAddress address the Bridgehub contract address
+  /// @param l2UptimeFeedAddr the address of the ZKSyncSequencerUptimeFeedInterface contract address
+  /// @param gasLimit the gasLimit to use for sending a message from L1 to L2
+  ///
+  constructor(
+    address l1CrossDomainMessengerAddress,
+    address l2UptimeFeedAddr,
+    uint32 gasLimit,
+    uint32 chainId,
+    uint32 l2GasPerPubdataByteLimit
+  ) {
+    // solhint-disable-next-line gas-custom-errors
+    require(l1CrossDomainMessengerAddress != address(0), "Invalid xDomain Messenger address");
+    // solhint-disable-next-line gas-custom-errors
+    require(l2UptimeFeedAddr != address(0), "Invalid ZKSyncSequencerUptimeFeedInterface contract address");
+    L1_CROSS_DOMAIN_MESSENGER_ADDRESS = l1CrossDomainMessengerAddress;
+    L2_UPTIME_FEED_ADDR = l2UptimeFeedAddr;
+    s_gasLimit = gasLimit;
+
+    // Check if the chainId is one of the valid values
+    require(chainId == TEST_NET_CHAIN_ID || chainId == MAIN_NET_CHAIN_ID, "Invalid chain id");
```

Comment:
Let's do this:

1. Let's define a custom error: `error InvalidChainID();` one line underneath `event GasLimitUpdated(...);`
2. Then, let's modify this code as follows so that this resolves the failing CI lint check:

```suggestion
    if (chainId != TEST_NET_CHAIN_ID && chainId != MAIN_NET_CHAIN_ID, "Invalid chain id") {
      revert InvalidChainID();
    }
```

Here's the relevant style guide section for reference:

https://github.com/smartcontractkit/chainlink/blob/develop/contracts/STYLE_GUIDE.md#expose-errors